core/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -99,6 +101,12 @@ public Settings additionalSettings() {
     public void onIndexModule(IndexModule indexModule) {}
 
     /**
+     * Called after a search response finishes executing, just prior to being sent back to the client.
+     * Can be used to collect details about the request/response
+     */
+    public void onFinishedSearchResponse(SearchRequest request, SearchResponse response) {}
```

Comment:
oh I don't think we should do this.. don't mix plugin inteface with stuff that is called at runtime? Plugin is a setup like inteface?